pkg/workload/workload.go

```diff
@@ -1112,15 +1116,19 @@ func resetClusterNomination(w *kueue.Workload) {
 	w.Status.NominatedClusterNames = nil
 }
 
-func reportEvictedWorkload(recorder record.EventRecorder, wl *kueue.Workload, cqName kueue.ClusterQueueReference, reason, message string) {
-	metrics.ReportEvictedWorkloads(cqName, reason)
+func reportEvictedWorkload(recorder record.EventRecorder, wl *kueue.Workload, cqName kueue.ClusterQueueReference, reason, underlyingCause, message string) {
+	metrics.ReportEvictedWorkloads(cqName, reason, underlyingCause)
 	if podsReadyToEvictionTime := workloadsWithPodsReadyToEvictedTime(wl); podsReadyToEvictionTime != nil {
-		metrics.PodsReadyToEvictedTimeSeconds.WithLabelValues(string(cqName), reason).Observe(podsReadyToEvictionTime.Seconds())
+		metrics.PodsReadyToEvictedTimeSeconds.WithLabelValues(string(cqName), reason, underlyingCause).Observe(podsReadyToEvictionTime.Seconds())
 	}
 	if features.Enabled(features.LocalQueueMetrics) {
-		metrics.ReportLocalQueueEvictedWorkloads(metrics.LQRefFromWorkload(wl), reason)
+		metrics.ReportLocalQueueEvictedWorkloads(metrics.LQRefFromWorkload(wl), reason, underlyingCause)
+	}
+	if reason == kueue.WorkloadDeactivated && underlyingCause != "" {
+		recorder.Event(wl, corev1.EventTypeNormal, fmt.Sprintf("%sDueTo%sDueTo%s", kueue.WorkloadEvicted, reason, underlyingCause), message)
+	} else {
+		recorder.Event(wl, corev1.EventTypeNormal, fmt.Sprintf("%sDueTo%s", kueue.WorkloadEvicted, reason), message)
 	}
```

Comment:
```suggestion
	eventReason := fmt.Sprintf("%sDueTo%s", kueue.WorkloadEvicted, reason)
	if reason == kueue.WorkloadDeactivated && underlyingCause != "" {
		eventReason = fmt.Sprintf("%sDueTo%s", eventReason, underlyingCause)
	}
	recorder.Event(wl, corev1.EventTypeNormal, eventReason, message)
```

Same comments as above.

Reply:
Changed here as well